### PR TITLE
Folio Changes 

### PR DESCRIPTION
**Pull request by midhun-aot**

### CODE REVIEW

File: backend/sites/src/app/resolvers/folio/folio.resolver.ts [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    @Args('folioDTO', { type: () => FolioMinDTO }, new ValidationPipe())
+    folioDTO: FolioMinDTO,
+    @AuthenticatedUser()
+    user: any
+  ) {
+
+    const result = await this.folioService.getSitesForFolio(folioDTO,user);
+
+    if (result.length > 0) {
+      return this.genericResponseProviderForFolioContent.createResponse('Folio fetched successfully', 200, true, result);
```
Comment (nupurdixit13) [inline backend/sites/src/app/resolvers/folio/folio.resolver.ts]: IMO the message should be `Sites fetched successfully for folio ${folioID}` since the method deals with getting sites for folio . Same holds true for the message in else section as well.

File: backend/sites/src/app/dto/Folio.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    @Field()
+    @IsString()
+    @IsOptional()    
+    whenUpdated?: string;
+}
+
+
+@InputType()
+export  class FolioMinDTO {
```
Comment (nupurdixit13) [inline backend/sites/src/app/dto/Folio.ts]: Can we please give a more meaningful name as `FolioMin` is not very self explanatory? Any thoughts?

File: backend/sites/src/app/dto/Folio.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    @Field()  
+    @IsNumber()   
+    id: number;
+    
+    @Field()  
+    @IsString()
+    userId: string;      
+   
+}
+
```
Comment (nupurdixit13) [inline backend/sites/src/app/dto/Folio.ts]: nit: Can we pls get rid of the extra spaces everywhere?

File: backend/sites/src/app/resolvers/cart/cart.resolver.ts [hunk 4]
```diff
@@ -42,29 +43,39 @@ export class CartResolver {
...
 
     if (message) {
-      return this.genericResponseProvider.createResponse(message, 201, true);
-    } else {
-      return this.genericResponseProvider.createResponse(
-        `Failed to add cart item. `,
-        400,
-        false,
-      );
+      return this.genericResponseProvider.createResponse("Deleted", 200, true);
```
Comment (nupurdixit13) [inline backend/sites/src/app/resolvers/cart/cart.resolver.ts]: Can we please have a descriptive message instead of 'Deleted'?

File: backend/sites/src/app/resolvers/cart/cart.resolver.ts [hunk 5]
```diff
@@ -42,29 +43,39 @@ export class CartResolver {
...
+    }
+    else
+    {
+      return this.genericResponseProvider.createResponse(`Failed delete item. `, 400, false);
```
Comment (nupurdixit13) [inline backend/sites/src/app/resolvers/cart/cart.resolver.ts]: Can we please reword the message here?

File: backend/sites/src/app/services/cart/cart.service.ts [hunk 6]
```diff
@@ -24,42 +24,63 @@ export class CartService {
...
+      for (const cartDTO of list) {
+        const { userId, siteId } = cartDTO;
 
-      if (recordExists) {
-        return 'Cart Item Already Exists';
-      } else {
-        // Convert the DTO to entity
-        const cartItem = plainToInstance(Cart, cartDTO);
-        const result = await this.cartRepository.save(cartItem);
+        // Check if the combination of userId and siteId exists in the table
```
Comment (nupurdixit13) [inline backend/sites/src/app/services/cart/cart.service.ts]: Curious as why we don't wanna show message in case record already exists.

File: backend/sites/src/app/services/cart/cart.service.ts [hunk 7]
```diff
@@ -24,42 +24,63 @@ export class CartService {
...
+        const existingRecord = await this.cartRepository.findOne({
+          where: { userId, siteId },
+        });
 
-        if (result) {
-          return 'Cart Item Added.';
+        if (!existingRecord) {       
+          // Convert the DTO to entity
+          const cartItem = plainToInstance(Cart, cartDTO);
+          const result = await this.cartRepository.save(cartItem);
```
Comment (nupurdixit13) [inline backend/sites/src/app/services/cart/cart.service.ts]: We need to check for `result` and decide whether the record was inserted successfully or not.

File: backend/sites/src/app/services/cart/cart.service.ts [hunk 8]
```diff
@@ -24,42 +24,63 @@ export class CartService {
...
+  async deleteCartItem(cartDeleteList: CartDeleteDTO[]): Promise<boolean> {
     try {
-      if (cartId != '') {
-        const result = await this.cartRepository.delete({ id: cartId });
-        if (result.affected > 0) return true;
-        else return false;
+      for (const item of cartDeleteList) {
+        const { cartId, userId } = item;
+        if (cartId != '') {
+          const result = await this.cartRepository.delete({
```
Comment (nupurdixit13) [inline backend/sites/src/app/services/cart/cart.service.ts]: Same as above, need to check for `result` to decide whether record was successfully deleted or not.

File: frontend/site-registry/src/app/features/folios/dto/Folio.ts [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+
+    constructor( siteId:string, userId:string, folioId:string, id:number, whoCreated:string)
+    {
+      this.userId = userId;
+      this.folioId = folioId;
+      this.siteId = siteId;
+      this.whoCreated = whoCreated;
+      this.id = id;
+    }
+
```
Comment (nupurdixit13) [inline frontend/site-registry/src/app/features/folios/dto/Folio.ts:62]: Pls remove spaces

File: backend/sites/src/app/services/folio/folioContents.service.ts [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+         
+        const result = await this.folioContentRepository.save(folioContent);
+
+        if (result) {
+          return true;
+        }
+      }
+    } catch (error) {
+      throw new Error('Failed to folio content');
+    }
```
Comment (nupurdixit13) [inline backend/sites/src/app/services/folio/folioContents.service.ts]: Can we pls reword it to  `Failed to add folio content`?. I guess you must have meant the same but just missed add keyword :)

File: backend/sites/src/app/services/folio/folioContents.service.ts [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+    const { siteId, folioId } = inputDTO;
+    try {
+      const existingRecord = await this.folioContentRepository.findOne({
+        where: { siteId, folioId },
+      });
+
+      if (existingRecord) {
+        return false;
+      } else {
+        delete inputDTO.id;
```
Comment (nupurdixit13) [inline backend/sites/src/app/services/folio/folioContents.service.ts]: Why is this deletion required, if the record is not existing we can straight add it, isn't?

File: backend/sites/src/app/services/folio/folioContents.service.ts [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+        else return false;
+      }
+
+      return false;
+    } catch (e) {
+      return false;
+    }
+  }
+
+  async deleteSitesInFolio(folioId: string, siteId: string ): Promise<boolean> {
```
Comment (nupurdixit13) [inline backend/sites/src/app/services/folio/folioContents.service.ts:84]: IMO the return type for this and other methods wherein only Boolean is stated should be `Promise<{ success: boolean, message?: string }` so that we know the exact reason behind the failure. Let me know your thoughts. Ignore message if success: true and have an error message in case success: false(failure)

File: backend/sites/src/app/services/folio/folioContents.service.ts [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+        else return false;
+      }
+
+      return false;
+    } catch (e) {
+      return false;
+    }
+  }
+
+  async deleteAllSitesInFolio(folioId: string): Promise<boolean> {
```
Comment (nupurdixit13) [inline backend/sites/src/app/services/folio/folioContents.service.ts:105]: Same as above

File: frontend/site-registry/src/app/helpers/utility.ts [hunk 14]
```diff
@@ -92,3 +92,32 @@ export const getAxiosInstance = () => {
 
   return instance;
 };
+
+
+export const deepSearch = (obj: any, searchTerm: string): boolean => {
```
Comment (nupurdixit13) [inline frontend/site-registry/src/app/helpers/utility.ts:99]: Can we add a description stating what it exactly does?

File: backend/sites/src/app/resolvers/folio/folio.resolver.ts (hunk 1 above)
Comment (midhun-aot) [reply backend/sites/src/app/resolvers/folio/folio.resolver.ts]: Done.

File: backend/sites/src/app/dto/Folio.ts (hunk 3 above)
Comment (midhun-aot) [reply backend/sites/src/app/dto/Folio.ts]: Done.

File: backend/sites/src/app/resolvers/cart/cart.resolver.ts (hunk 4 above)
Comment (midhun-aot) [reply backend/sites/src/app/resolvers/cart/cart.resolver.ts]: We don't show this to User in frontend. Its just returning some message as part of response. Anyway I have updated to me more descriptive.

File: backend/sites/src/app/resolvers/cart/cart.resolver.ts (hunk 5 above)
Comment (midhun-aot) [reply backend/sites/src/app/resolvers/cart/cart.resolver.ts]: done.

File: backend/sites/src/app/services/cart/cart.service.ts (hunk 6 above)
Comment (midhun-aot) [reply backend/sites/src/app/services/cart/cart.service.ts]: Currently we are loop through an  sites array. So if user selected already an existing site we are not adding it again to cart. But not showing any message right now, ( as notification part was not implemented that time)  but can make it happen.

File: backend/sites/src/app/services/folio/folioContents.service.ts (hunk 10 above)
Comment (midhun-aot) [reply backend/sites/src/app/services/folio/folioContents.service.ts]: done

File: frontend/site-registry/src/app/features/folios/dto/Folio.ts (hunk 9 above)
Comment (midhun-aot) [reply frontend/site-registry/src/app/features/folios/dto/Folio.ts:62]: done

File: backend/sites/src/app/services/folio/folioContents.service.ts (hunk 11 above)
Comment (midhun-aot) [reply backend/sites/src/app/services/folio/folioContents.service.ts]: i added it because I wanted to delete the property before passing to plainToInstance. anyway I have update the code and no longer using it.

File: backend/sites/src/app/services/folio/folioContents.service.ts (hunk 12 above)
Comment (midhun-aot) [reply backend/sites/src/app/services/folio/folioContents.service.ts:84]: The delete function delete the record if it find the match record where clause other than that there is no reason. So returning true or false which shows deleted or not

File: backend/sites/src/app/services/cart/cart.service.ts (hunk 6 above)
Comment (midhun-aot) [reply backend/sites/src/app/services/cart/cart.service.ts]: Update code to have bulk update.

File: backend/sites/src/app/services/cart/cart.service.ts (hunk 8 above)
Comment (midhun-aot) [reply backend/sites/src/app/services/cart/cart.service.ts]: done.

File: backend/sites/src/app/services/cart/cart.service.ts (hunk 7 above)
Comment (midhun-aot) [reply backend/sites/src/app/services/cart/cart.service.ts]: Updated code.

File: backend/sites/src/app/dto/Folio.ts [hunk 15]
```diff
@@ -0,0 +1,60 @@
+import { Field, InputType, ObjectType } from '@nestjs/graphql';
+import { FolioContentDTO } from './folioContent';
```
Comment (nikhila-aot) [inline backend/sites/src/app/dto/Folio.ts]: have consistent naming for filename please

File: backend/sites/src/app/resolvers/cart/cart.resolver.ts [hunk 16]
```diff
@@ -20,60 +30,125 @@ export class CartResolver {
...
+      if (result)
+        return this.genericResponseProvider.createResponse(
+          'Items added to cart',
+          201,
+          true,
+        );
+      else
+        return this.genericResponseProvider.createResponse(
+          'Unable to add items to cart',
+          200,
```
Comment (nikhila-aot) [inline backend/sites/src/app/resolvers/cart/cart.resolver.ts]: wrong error code here

File: backend/sites/src/app/resolvers/cart/cart.resolver.ts [hunk 17]
```diff
@@ -20,60 +30,125 @@ export class CartResolver {
...
+        return this.genericResponseProvider.createResponse(
+          'Cart items fetched successfully',
+          200,
+          true,
+          result,
+        );
+      } else {
+        return this.genericResponseProvider.createResponse(
+          `Cart items not found for user id: ${userId}`,
+          200,
```
Comment (nikhila-aot) [inline backend/sites/src/app/resolvers/cart/cart.resolver.ts]: wrong error code used here

File: backend/sites/src/app/resolvers/cart/cart.resolver.ts [hunk 18]
```diff
@@ -20,60 +30,125 @@ export class CartResolver {
...
-        false,
+    try {
+      const message = await this.cartService.deleteCartItem(
+        cartDeleteDTO,
+        user?.sub,
       );
+
+      if (message) {
+        return this.genericResponseProvider.createResponse(
+          'Succefully deleted cart items.',
```
Comment (nikhila-aot) [inline backend/sites/src/app/resolvers/cart/cart.resolver.ts]: typo for Successfully

File: backend/sites/src/app/resolvers/folio/folio.resolver.ts [hunk 19]
```diff
@@ -0,0 +1,231 @@
...
+    @Args('userId', { type: () => String }) userId: string,
+    @AuthenticatedUser()
+    user: any
+  ) {
+
+    try {
+
+      const result = await this.folioService.getFoliosForUser(user);
+
+      if (result.length > 0) {
```
Comment (nikhila-aot) [inline backend/sites/src/app/resolvers/folio/folio.resolver.ts]: you will have to check if result has content before using result.length

File: backend/sites/src/app/resolvers/folio/folio.resolver.ts [hunk 20]
```diff
@@ -0,0 +1,231 @@
...
+        return this.genericResponseProvider.createResponse('Folio fetched successfully', 200, true, result);
+      }
+      else
+      {
+        return this.genericResponseProvider.createResponse(`Folio not found for user id: ${userId}`, 200, true, []);
```
Comment (nikhila-aot) [inline backend/sites/src/app/resolvers/folio/folio.resolver.ts:40]: wrong error code used

File: backend/sites/src/app/resolvers/folio/folio.resolver.ts [hunk 21]
```diff
@@ -0,0 +1,231 @@
...
+    folioDTO: FolioMinDTO,
+    @AuthenticatedUser()
+    user: any
+  ) {
+
+    try {
+
+      const result = await this.folioService.getSitesForFolio(folioDTO,user);
+
+      if (result.length > 0) {
```
Comment (nikhila-aot) [inline backend/sites/src/app/resolvers/folio/folio.resolver.ts]: check if result is empty and then do result.length

File: backend/sites/src/app/resolvers/folio/folio.resolver.ts [hunk 22]
```diff
@@ -0,0 +1,231 @@
...
+        return this.genericResponseProviderForFolioContent.createResponse('Sites fetched successfully for folio', 200, true, result);
+      }
+      else
+      {
+        return this.genericResponseProviderForFolioContent.createResponse(`Unable to fetch sites for folio`, 200, true, []);
```
Comment (nikhila-aot) [inline backend/sites/src/app/resolvers/folio/folio.resolver.ts:71]: wrong error code

File: backend/sites/src/app/services/folio/folio.service.ts [hunk 23]
```diff
@@ -0,0 +1,202 @@
...
+  }
+
+  async getSitesForFolio(
+    input: FolioMinDTO,
+    userInfo: any,
+  ): Promise<FolioContents[] | null> {
+    try {
+      const userId = userInfo.sub;
+      const { id } = input;
+      const folio = await this.folioRepository.findOne({
```
Comment (nikhila-aot) [inline backend/sites/src/app/services/folio/folio.service.ts:42]: can we have have the return object 'folio' is handled consistent across both the methods getFoliosForUser and getSitesForFolio

File: backend/sites/src/app/services/folio/folio.service.ts [hunk 24]
```diff
@@ -0,0 +1,202 @@
...
+        return false;
+      } else {
+        const folio = plainToInstance(Folio, inputDTO);
+        folio.whenCreated = new Date();
+        folio.whenUpdated = new Date();
+        const result = await this.folioRepository.save(folio);
+
+        if (result) return true;
+
+        return false;
```
Comment (nikhila-aot) [inline backend/sites/src/app/services/folio/folio.service.ts]: do we return false from both the if and else blocks ?

File: backend/sites/src/app/services/folio/folio.service.ts [hunk 25]
```diff
@@ -0,0 +1,202 @@
...
+    try {
+      const userId = userInfo.sub;
+      const updatePromises = inputArr.map(async (inputDTO) => {
+        const { id, description, folioId } = inputDTO;
+
+        const existingRecord = await this.folioRepository.findOne({
+          where: { id, userId },
+        });
+
+        if (existingRecord) {
```
Comment (nikhila-aot) [inline backend/sites/src/app/services/folio/folio.service.ts:99]: we should have an else block for this if()

File: backend/sites/src/app/services/folio/folio.service.ts [hunk 26]
```diff
@@ -0,0 +1,202 @@
...
+      await Promise.all(updatePromises);
+    } catch (error) {
+      console.error('Error', error);
+      throw new Error('Failed to update folio content');
+    }
+
+    return true;
+  }
+
+  async deleteFolio(id: number, userInfo: any): Promise<boolean> {
```
Comment (nikhila-aot) [inline backend/sites/src/app/services/folio/folio.service.ts:125]: is this the folio id? can we rename it ?

File: backend/sites/src/app/services/folio/folio.service.ts [hunk 27]
```diff
@@ -0,0 +1,202 @@
...
+        return false;
+      }
+
+      const userId = userInfo.sub;
+
+      const savedFolio = this.folioRepository.findOne({
+        where: { id, userId },
+      });
+
+      if (savedFolio != null) {
```
Comment (nikhila-aot) [inline backend/sites/src/app/services/folio/folio.service.ts:137]: have an else block for this

File: backend/sites/src/app/services/folio/folio.service.ts [hunk 28]
```diff
@@ -0,0 +1,202 @@
...
+      return false;
+    } catch (error) {
+      console.error('Error in deleteFolio:', error);
+      return false;
+    }
+  }
+
+  async addSiteToFolio(
+    inputDTO: [FolioContentDTO],
+    userInfo: any,
```
Comment (nikhila-aot) [inline backend/sites/src/app/services/folio/folio.service.ts:156]: can we use the same bulk insert here ?

File: backend/sites/src/app/services/folio/folio.service.ts [hunk 29]
```diff
@@ -0,0 +1,202 @@
...
+    } catch (error) {
+      console.error('Error', error);
+      throw error;
+    }
+  }
+
+  async deleteSitesInFolio(
+    inputDTO: [FolioContentDTO],
+    userInfo: any,
+  ): Promise<boolean> {
```
Comment (nikhila-aot) [inline backend/sites/src/app/services/folio/folio.service.ts:186]: can we use bulk delete ?

File: frontend/site-registry/src/app/features/folios/FolioTableConfig.tsx [hunk 30]
```diff
@@ -0,0 +1,170 @@
...
+    id: 3,
+    displayName: "Last Updated",
+    active: true,
+    graphQLPropertyName: "whenUpdated",
+    displayType: {
+      label: "",
+      type: FormFieldType.Label,
+      tableMode:true,
+      graphQLPropertyName: "whenUpdated",
+      // customLabelCss:'custom-participant-lbl-text',
```
Comment (nikhila-aot) [inline frontend/site-registry/src/app/features/folios/FolioTableConfig.tsx]: can you remove all commented code

File: backend/sites/src/app/services/folio/folioContents.service.ts [hunk 31]
```diff
@@ -0,0 +1,108 @@
...
+
+  async getSiteForFolio(folioId: string): Promise<FolioContents[]> {
+    try {
+      const sitesForFolioArr = await this.folioContentRepository.find({  
+        relations: ['site','folio'],
+        where: { folioId },
+      });
+      return sitesForFolioArr;
+    } catch (error) {
+      throw error;
```
Comment (nikhila-aot) [inline backend/sites/src/app/services/folio/folioContents.service.ts:25]: can we have consistency in all service layer error handling ?

File: backend/sites/src/app/services/folio/folioContents.service.ts [hunk 32]
```diff
@@ -0,0 +1,108 @@
...
+        const folioContent = {
+          siteId: inputDTO.siteId,
+          folioId : inputDTO.folioId,
+          whoCreated: inputDTO.whoCreated,
+          userId: inputDTO.userId
+        }
+         
+        const result = await this.folioContentRepository.save(folioContent);
+
+        if (result) {
```
Comment (nikhila-aot) [inline backend/sites/src/app/services/folio/folioContents.service.ts:50]: else block for this ?

File: backend/sites/src/app/services/folio/folioContents.service.ts [hunk 33]
```diff
@@ -0,0 +1,108 @@
...
+      }
+    } catch (error) {
+      throw new Error('Failed to add folio content');
+    }
+  }
+
+
+  async deleteFolioContent(folioContentId: string): Promise<boolean> {
+    try {
+      if (folioContentId != '') {
```
Comment (nikhila-aot) [inline backend/sites/src/app/services/folio/folioContents.service.ts]: can this be null or empty ?

File: backend/sites/src/app/services/folio/folioContents.service.ts [hunk 34]
```diff
@@ -0,0 +1,108 @@
...
+
+      return false;
+    } catch (e) {
+      return false;
+    }
+  }
+
+  async deleteSitesInFolio(folioId: string, siteId: string ): Promise<boolean> {
+    try {
+      if (folioId && siteId) {
```
Comment (nikhila-aot) [inline backend/sites/src/app/services/folio/folioContents.service.ts]: else block for this ?

File: backend/sites/src/app/services/folio/folioContents.service.ts [hunk 35]
```diff
@@ -0,0 +1,108 @@
...
+      return false;
+    } catch (e) {
+      return false;
+    }
+  }
+
+  async deleteAllSitesInFolio(folioId: string): Promise<boolean> {
+    try {
+      
+      if (folioId) {
```
Comment (nikhila-aot) [inline backend/sites/src/app/services/folio/folioContents.service.ts]: have an else block pls

File: backend/sites/src/app/resolvers/cart/cart.resolver.ts (hunk 16 above)
Comment (midhun-aot) [reply backend/sites/src/app/resolvers/cart/cart.resolver.ts]: done

File: backend/sites/src/app/dto/Folio.ts (hunk 15 above)
Comment (midhun-aot) [reply backend/sites/src/app/dto/Folio.ts]: done

File: backend/sites/src/app/resolvers/cart/cart.resolver.ts (hunk 18 above)
Comment (midhun-aot) [reply backend/sites/src/app/resolvers/cart/cart.resolver.ts]: done

File: backend/sites/src/app/resolvers/folio/folio.resolver.ts (hunk 20 above)
Comment (midhun-aot) [reply backend/sites/src/app/resolvers/folio/folio.resolver.ts:40]: done

File: backend/sites/src/app/resolvers/folio/folio.resolver.ts (hunk 21 above)
Comment (midhun-aot) [reply backend/sites/src/app/resolvers/folio/folio.resolver.ts]: done

File: backend/sites/src/app/services/folio/folio.service.ts (hunk 24 above)
Comment (midhun-aot) [reply backend/sites/src/app/services/folio/folio.service.ts]: done, updated the logic.

File: backend/sites/src/app/services/folio/folio.service.ts (hunk 25 above)
Comment (midhun-aot) [reply backend/sites/src/app/services/folio/folio.service.ts:99]: done

File: backend/sites/src/app/services/folio/folio.service.ts (hunk 26 above)
Comment (midhun-aot) [reply backend/sites/src/app/services/folio/folio.service.ts:125]: done

File: backend/sites/src/app/services/folio/folio.service.ts (hunk 29 above)
Comment (midhun-aot) [reply backend/sites/src/app/services/folio/folio.service.ts:186]: No delete from this block , separate service for folio content invoked from here.

File: backend/sites/src/app/services/folio/folioContents.service.ts (hunk 35 above)
Comment (midhun-aot) [reply backend/sites/src/app/services/folio/folioContents.service.ts]: done

File: backend/sites/src/app/services/folio/folioContents.service.ts (hunk 34 above)
Comment (midhun-aot) [reply backend/sites/src/app/services/folio/folioContents.service.ts]: done

File: backend/sites/src/app/services/folio/folioContents.service.ts (hunk 33 above)
Comment (midhun-aot) [reply backend/sites/src/app/services/folio/folioContents.service.ts]: done

File: backend/sites/src/app/services/folio/folioContents.service.ts (hunk 32 above)
Comment (midhun-aot) [reply backend/sites/src/app/services/folio/folioContents.service.ts:50]: done.

File: backend/sites/src/app/services/folio/folioContents.service.ts (hunk 31 above)
Comment (midhun-aot) [reply backend/sites/src/app/services/folio/folioContents.service.ts:25]: done

File: frontend/site-registry/src/app/features/folios/FolioTableConfig.tsx (hunk 30 above)
Comment (midhun-aot) [reply frontend/site-registry/src/app/features/folios/FolioTableConfig.tsx]: done

File: backend/sites/src/app/services/folio/folio.service.ts (hunk 28 above)
Comment (midhun-aot) [reply backend/sites/src/app/services/folio/folio.service.ts:156]: calling folio content service inside,

File: backend/sites/src/app/services/folio/folio.service.ts [hunk 36]
```diff
@@ -0,0 +1,210 @@
...
+
+      if (!existingRecord) 
+      {
+        const folio = plainToInstance(Folio, inputDTO);
+        folio.whenCreated = new Date();
+        folio.whenUpdated = new Date();
+        const result = await this.folioRepository.save(folio);
+
+        if (result) return true;
+      }
```
Comment (nikhila-aot) [inline backend/sites/src/app/services/folio/folio.service.ts:81]: have the else block

File: backend/sites/src/app/services/cart/cart.service.ts [hunk 37]
```diff
@@ -24,42 +28,100 @@ export class CartService {
...
-          return 'Cart Item Added.';
+        const recordKey = `${userId}-${siteId}`;
+        if (!existingRecordSet.has(recordKey)) {
+          const cartItem = plainToInstance(Cart, cartDTO);
+          cartItemsToInsert.push(cartItem);
         }
       }
+
+      let insertedRecords: Cart[] = [];
+      if (cartItemsToInsert.length > 0) {
```
Comment (nikhila-aot) [inline backend/sites/src/app/services/cart/cart.service.ts]: Can cartItemsToInsert be null or empty? If yes, pls handle it

File: backend/sites/src/app/services/cart/cart.service.ts [hunk 38]
```diff
@@ -24,42 +28,100 @@ export class CartService {
...
+        insertedRecords = await this.cartRepository.save(cartItemsToInsert);
+
+        if (insertedRecords.length > 0) return true;
```
Comment (nikhila-aot) [inline backend/sites/src/app/services/cart/cart.service.ts]: have if else block

File: backend/sites/src/app/services/cart/cart.service.ts [hunk 39]
```diff
@@ -24,42 +28,100 @@ export class CartService {
...
+      if (cartIds.length > 0) {
+        deleteResult =  await this.cartRepository
+          .createQueryBuilder()
+          .delete()
+          .from(Cart) 
+          .where('id IN (:...cartIds)', { cartIds })
+          .andWhere('userId = :userId', { userId: userId }) 
+          .execute();
+        
+        if(deleteResult.affected > 0) return true;
```
Comment (nikhila-aot) [inline backend/sites/src/app/services/cart/cart.service.ts]: have if else block

File: backend/sites/src/app/services/cart/cart.service.ts [hunk 40]
```diff
@@ -24,42 +28,100 @@ export class CartService {
...
+      if (siteIds.length > 0) {
+        deleteResult =  await this.cartRepository
+          .createQueryBuilder()
+          .delete()
+          .from(Cart) 
+          .where('siteId IN (:...cartIds)', { siteIds })
+          .andWhere('userId = :userId', { userId: userId }) 
+          .execute();
+        
+        if(deleteResult.affected > 0) return true;
```
Comment (nikhila-aot) [inline backend/sites/src/app/services/cart/cart.service.ts:124]: have if else block

File: backend/sites/src/app/services/folio/folio.service.ts (hunk 36 above)
Comment (midhun-aot) [reply backend/sites/src/app/services/folio/folio.service.ts:81]: done

File: backend/sites/src/app/services/cart/cart.service.ts (hunk 38 above)
Comment (midhun-aot) [reply backend/sites/src/app/services/cart/cart.service.ts]: done

File: backend/sites/src/app/services/cart/cart.service.ts (hunk 40 above)
Comment (midhun-aot) [reply backend/sites/src/app/services/cart/cart.service.ts:124]: done

File: backend/sites/src/app/services/cart/cart.service.ts (hunk 37 above)
Comment (midhun-aot) [reply backend/sites/src/app/services/cart/cart.service.ts]: done

**Review by nikhila-aot — APPROVED** [review]
